### ISSUE TITLE
Internal path alias working even when importing external, is it possible?

### ISSUE THREAD

**Issue by henriqueweiand** [NiGhTTraX/ts-monorepo#28]
Thank you in advance! your repository is helping me to learn a lot!

I want to bring a problem related to path alias between packages. I don't know if it's okay to do it the way I'm trying but come on, let me explain. I'm managing to call the packages between projects, so far so good. However, when I call one another and this other in his imports uses some alias (from the project itself) he gets lost.

![Screen Shot 2020-11-22 at 20 19 25](https://user-images.githubusercontent.com/6539492/99920155-643e8180-2d00-11eb-82c1-99f0fe649268.png)
 
In the common scenario, I would do a simple `import {Container, Input as InputBase, Error} from '. / Styles'`, but I want to try to reach that sense of different projects having their alias declared for them and in this external relationship with internal, worked. You see?

I brought an example of a POC that I'm doing where I had this problem that I reported ... I tried to apply the same problem in your project and I had the same error

https://github.com/henriqueweiand/poc-yarn-workspaces

**Comment by NiGhTTraX** [comment]
Hi, @henriqueweiand. I checked your linked repo and I saw that the `proxy` package doesn't have any aliases defined. Any package that wants to use aliases (in this case `uikit`) must have them defined in the `tsconfig.json` it's **built with**. `uikit` has the aliased defined in `tsconfig.paths.json`, but it's never built as a standalone package, it's built either as part of `auth` (which has the aliased defined) or as part of `proxy` (which doesn't).

<details>

<summary>By adding the necessary aliases to the proxy package you can then use aliases inside uikit.</summary>

```patch
Index: packages/libs/uikit/src/components/input/index.tsx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/libs/uikit/src/components/input/index.tsx b/packages/libs/uikit/src/components/input/index.tsx
--- a/packages/libs/uikit/src/components/input/index.tsx	(revision 47e9cb331e4d7b5c4897a63c7e4d9c38f2a7132d)
+++ b/packages/libs/uikit/src/components/input/index.tsx	(date 1606118345764)
@@ -1,6 +1,7 @@
 import React, { PropsWithChildren } from 'react'
 import { Container, Input as InputBase, Error } from './styles'
 import { InputProps } from './types'
+import 'components/input/styles';
 
 const Input: React.FC<PropsWithChildren<InputProps>> = ({
 	error,
Index: packages/apps/proxy/tsconfig.paths.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/apps/proxy/tsconfig.paths.json b/packages/apps/proxy/tsconfig.paths.json
new file mode 100644
--- /dev/null	(date 1606125350038)
+++ b/packages/apps/proxy/tsconfig.paths.json	(date 1606125350038)
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+      "baseUrl": "../../",
+      "paths": {
+      "components/*": ["./libs/uikit/src/components/*"]
+    }
+  }
+}
Index: packages/apps/proxy/tsconfig.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/apps/proxy/tsconfig.json b/packages/apps/proxy/tsconfig.json
--- a/packages/apps/proxy/tsconfig.json	(revision 47e9cb331e4d7b5c4897a63c7e4d9c38f2a7132d)
+++ b/packages/apps/proxy/tsconfig.json	(date 1606124899797)
@@ -1,4 +1,5 @@
 {
+	"extends": "./tsconfig.paths.json",
 	"compilerOptions": {
 		"target": "es5",
 		"lib": ["dom", "dom.iterable", "esnext"],
@@ -13,7 +14,6 @@
 		"resolveJsonModule": true,
 		"isolatedModules": true,
 		"jsx": "preserve",
-		"baseUrl": "."
 	},
 	"include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
 	"exclude": ["node_modules"]
```

</details>

----

P.S.: the monorepo package structure looks a bit weird to me. `auth` and `proxy` look like standalone projects, but `proxy` is importing pages from `auth` (which then imports `uikit`, hence the need to have the aliases defined in both projects). I would try to avoid dependencies between standalone projects and move the common stuff into libraries which are then depended on by all projects.

![image](https://user-images.githubusercontent.com/485061/99950495-e2d90480-2d7c-11eb-832f-bb5c6ad4c5f5.png)

The highlighted area above means that a single `tsconfig.json` will be used to compile all of that code and if the individual packages inside want to use aliases they all need to be defined in the root config.

**Comment by henriqueweiand** [comment]
The proxy, it is a case apart from tests ... but in general its explanation was super clear and gave me a good help! I made some changes to the repo that sent the link and it worked ...

I centralized the path alias all in the root and made the link in addition to a small change in the craco towards the babel. It was beautiful!

Thanks for your great explanation of the context ... maybe something from our conversation could actually be useful for the repo here too.
